runtime/lua/vim/lsp.lua

```diff
@@ -244,17 +247,29 @@ function lsp.start(config, opts)
   local reuse_client = opts.reuse_client or reuse_client_default
   local bufnr = resolve_bufnr(opts.bufnr)
 
+  if vim.bo[bufnr].buftype == 'nofile' or not vim.api.nvim_buf_is_loaded(bufnr) then
```

Comment:
```suggestion
  if not vim.api.nvim_buf_is_loaded(bufnr) then
```

As mentioned in the comments - there are currently working `nofile` use-cases

Reply:
Ah I completely misinterpreted your [comment](https://github.com/neovim/neovim/pull/28478#discussion_r1578000992) as "`nofile` definitely works [as the value to use in this PR]". My mistake.

Reply:
Ok given that `'buftype'` is not in use at all I'm just going to drop the first commit and make this PR only adding the `silent` option (since `nvim_buf_is_loaded` is already checked in `buf_attach_client`).